apps/desktop/src/routes/[projectId]/branch/[...name]/+page.svelte

```diff
@@ -1,65 +1,36 @@
 <script lang="ts">
-	import { Project } from '$lib/backend/projects';
-	import { BranchListingService } from '$lib/branches/branchListing';
 	import BranchPreview from '$lib/components/BranchPreview.svelte';
-	import FullviewLoading from '$lib/components/FullviewLoading.svelte';
 	import { getForgeListingService } from '$lib/forge/interface/forgeListingService';
 	import { RemoteBranchService } from '$lib/stores/remoteBranches';
-	import { groupBy } from '$lib/utils/groupBy';
-	import { error } from '$lib/utils/toasts';
 	import { Branch } from '$lib/vbranches/types';
 	import { getContext } from '@gitbutler/shared/context';
-	import { goto } from '$app/navigation';
 	import { page } from '$app/stores';
 
-	const project = getContext(Project);
-	const branchListingService = getContext(BranchListingService);
-	const branchListings = branchListingService.branchListings;
-
 	const remoteBranchService = getContext(RemoteBranchService);
-	const branches = remoteBranchService.branches;
-	const branchesByGivenName = $derived(groupBy($branches, (branch) => branch.givenName));
-
-	const branchListing = $derived($branchListings.find((bl) => bl.name === $page.params.name));
 
 	const forgeListingService = getForgeListingService();
+	const name = $derived($page.params.name);
 	const prs = $derived($forgeListingService?.prs);
-	const pr = $derived($prs?.find((pr) => pr.sourceBranch === branchListing?.name));
+	const pr = $derived($prs?.find((pr) => pr.sourceBranch === name));
 
 	let localBranch = $state<Branch>();
 	let remoteBranches = $state<Branch[]>([]);
 
 	$effect(() => {
-		if (branchListing) {
-			if (branchListing.virtualBranch?.inWorkspace) {
-				goto(`/${project.id}/board`);
-				return;
-			}
-
-			const branchesWithGivenName: Branch[] | undefined = branchesByGivenName[branchListing.name];
-
-			if (branchesWithGivenName) {
-				localBranch = branchesWithGivenName.find((branch) => !branch.isRemote);
-
-				remoteBranches = branchesWithGivenName.filter((branch) => branch.isRemote);
-			} else {
-				error('Failed to find branch');
-				goto(`/${project.id}/board`);
-			}
-		}
+		if (!name) return;
+		remoteBranchService.findBranches(name).then((branches) => {
+			localBranch = branches.find((branch) => !branch.isRemote);
+			remoteBranches = branches.filter((branch) => branch.isRemote);
+		});
 	});
 </script>
 
-{#if $branchListings.length === 0}
-	<FullviewLoading />
-{:else if branchListing}
-	{#if remoteBranches.length === 0 && localBranch}
-		<BranchPreview {localBranch} {pr} />
-	{:else}
-		{#each remoteBranches as remoteBranch}
-			<BranchPreview {localBranch} {remoteBranch} {pr} />
-		{/each}
-	{/if}
-{:else}
+{#if localBranch}
+	<BranchPreview {localBranch} {pr} />
+{/if}
+{#each remoteBranches as remoteBranch}
+	<BranchPreview {remoteBranch} {pr} />
+{/each}
```

Comment:
Is there a particular design wise reason we're changing thew way we display Local and Remote previews?

Reply:
Right this is perhaps a mistake on my part. Can you confirm we do _not_ want to show the local branch if one or more remote branches exist?

Reply:
I'll add back a loading state as well.

Reply:
> Can you confirm we do not want to show the local branch if one or more remote branches exist?
Yes. If both a local and remote branch is provided to the component, it displays the local branch commits as the branch commits, and the remote commits as upstream ones; like how it is in the workspace